Userland/Libraries/LibWeb/Painting/SVGMaskable.cpp

```diff
@@ -71,42 +71,44 @@ Optional<Gfx::Bitmap::MaskKind> SVGMaskable::get_mask_type_of_svg() const
     return {};
 }
 
-RefPtr<Gfx::Bitmap> SVGMaskable::calculate_mask_of_svg(PaintContext& context, CSSPixelRect const& masking_area) const
+OwnPtr<DisplayList> SVGMaskable::calculate_mask_of_svg(PaintContext& context, CSSPixelRect const& masking_area) const
 {
     auto const& graphics_element = verify_cast<SVG::SVGGraphicsElement const>(*dom_node_of_svg());
     auto mask_rect = context.enclosing_device_rect(masking_area);
-    auto paint_mask_or_clip = [&](PaintableBox const& paintable) -> RefPtr<Gfx::Bitmap> {
-        auto mask_bitmap_or_error = Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, mask_rect.size().to_type<int>());
-        RefPtr<Gfx::Bitmap> mask_bitmap = {};
-        if (mask_bitmap_or_error.is_error())
-            return {};
-        mask_bitmap = mask_bitmap_or_error.release_value();
-        DisplayList display_list;
-        DisplayListRecorder display_list_recorder(display_list);
+
+    auto paint_mask_or_clip = [&](PaintableBox const& paintable) -> OwnPtr<DisplayList> {
+        auto display_list = make<DisplayList>();
+        display_list->set_bitmap_rect(mask_rect);
```

Comment:
Storing this in the DisplayList seems janky. Is there a way to calculate the rect from the items in the DisplayList?

Reply:
nope, but you can add one! should be easy enough, because a display list command already know its bounding box.

Reply:
So, something like the maximum width and height of all commands' bounding boxes? I'm not getting quite the right number yet.